### PR TITLE
Allow 429 response code to trigger throttle detection

### LINKED ISSUE #1850: Check status code for 429 for throttling_error?
https://github.com/aws/aws-sdk-ruby/blob/1a86b112d8c76f23a1f2be8ee3df1432ba821418/gems/aws-sdk-core/lib/aws-sdk-core/plugins/retry_errors.rb#L106-L108

`throttling_error?` should also check the status code for `429` to determine `throttling_error?` as well

Potential New Code:
```ruby
def throttling_error?
  !!(THROTTLING_ERRORS.include?(@name) || @name.match(/throttl/i) || @http_status_code == 429)
end
```

*Not a Contribution*

### PR DESCRIPTION
**Pull request by arianf**
This closes #1850 

Allow AWS to detect a 429 response code as a throttle error instead of just relying on regex matching.

Current code relies on regex matching:
https://github.com/aws/aws-sdk-ruby/blob/1a86b112d8c76f23a1f2be8ee3df1432ba821418/gems/aws-sdk-core/lib/aws-sdk-core/plugins/retry_errors.rb#L106-L108

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429

> The HTTP 429 `Too Many Requests` response status code indicates the user has sent too many requests in a given amount of time ("rate limiting").

> 429 Too Many Requests

Also see, https://tools.ietf.org/html/rfc6585#section-4

### CODE REVIEW

File: gems/aws-eventstream/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 Unreleased Changes
 ------------------
 
+* Feature - Allow 429 response code to trigger throttle detection
```
Comment (awood45) [inline gems/aws-eventstream/CHANGELOG.md]: This would need to be in aws-sdk-core, as far as I can see you have not modified the eventstream gem.

**Comment by awood45** [comment]
Also, before I can merge please confirm: "By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license."

**Comment by awood45** [comment]
Just need the confirmation on the licensing and we're good to go.

**Comment by arianf** [comment]
Sorry for the delay, had to verify it was okay for me to make this statement.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

**Comment by awood45** [comment]
No worries! This will go out with our next release.

**Pull request merged**